lib/plugins/threaded-signing/plugin.c

```diff
@@ -254,31 +319,39 @@ threaded_openssl_get_signature(sv_threaded_plugin_t *self,
   SignedVideoReturnCode status = SV_OK;
 
   g_mutex_lock(&self->mutex);
-  if (self->plugin_state == THREADED_SIGNING_HAS_SIGNATURE && self->signature_info) {
-    if (self->signature_info->signature_size > max_signature_size) {
-      // If there is no room to copy the signature, report zero size.
-      *written_signature_size = 0;
-    } else {
-      memcpy(signature, self->signature_info->signature, self->signature_info->signature_size);
-      *written_signature_size = self->signature_info->signature_size;
-    }
-    // Change state and mark as copied.
-    self->plugin_state = THREADED_SIGNING_WAITS_FOR_HASH_TO_SIGN;
-    has_copied_signature = true;
-  } else if (self->plugin_state == THREADED_SIGNING_ERROR) {
-    *written_signature_size = 0;
-    // Change state and mark as copied.
-    self->plugin_state = THREADED_SIGNING_WAITS_FOR_HASH_TO_SIGN;
-    has_copied_signature = true;
+
+  if (!self->is_running) {
+    // Thread is not running. Go to done and return error.
+    status = SV_EXTERNAL_ERROR;
+    goto done;
+  }
+
+  if (self->output_buffer_idx == 0) goto done;
+
+  *written_signature_size = 0;
```

Comment:
This should be set before the sanity checks above.
Can even be moved before the lock.